### PR TITLE
parser: Use a lookup table for Delimiter::from_byte.

### PR DESCRIPTION
**Pull request by emilio**
It's faster.

### CODE REVIEW

**Comment by tiaanl** [comment]
@emilio Did you measure the performance of this change somewhere?

**Comment by emilio** [comment]
@tiaanl Our similar `match_byte` optimization is a huge perf improvement in the tokenizer, see #128. So I didn't measure this function explicitly but I'd expect similar gains.

**Comment by emilio** [comment]
Built a benchmark for this because I had some time. Before:

```
test tests::delimiter_from_byte ... bench:     177,361 ns/iter (+/- 7,390)
```

After:

```
test tests::delimiter_from_byte ... bench:      46,917 ns/iter (+/- 906)
```

**Review by tiaanl — CHANGES_REQUESTED** [review]
I was a bit skeptical about this one. It was either a few conditional jump instructions or a few memory accesses (the buffer doesn't fit fully into a cache line) and I was betting the memory accesses would take longer.

Benchmark numbers look like a good improvement.

File: src/tests.rs [hunk 1]
```diff
@@ -922,6 +922,18 @@ impl<'a> ToJson for CowRcStr<'a> {
     }
 }
 
+#[bench]
+#[cfg(feature = "bench")]
+fn delimiter_from_byte(b: &mut Bencher) {
+    b.iter(|| {
+        for _ in 0..1000 {
+            for i in 0..255 {
```
[the remark this answers is not in the document]
Comment (emilio) [reply src/tests.rs:930]: Indeed.

File: src/parser.rs [hunk 2]
```diff
@@ -313,16 +313,22 @@ impl Delimiters {
     }
 
     #[inline]
-    fn from_byte(byte: Option<u8>) -> Delimiters {
+    pub(crate) fn from_byte(byte: Option<u8>) -> Delimiters {
+        const TABLE: [Delimiters; 255] = {
```
[the remark this answers is not in the document]
Comment (emilio) [reply src/parser.rs:317]: Whoops, thanks!

**Comment by emilio** [comment]
Updated, thanks for the review @tiaanl!

**Review by tiaanl — APPROVED** [review]

**Pull request merged**